Sources/Ledger/TxOutUtils.swift

```diff
@@ -59,40 +36,92 @@ enum TxOutUtils {
+                        // RistrettoPrivate on success.
+                        return RistrettoPrivate(skippingValidation: bytes)
+                    case .failure(let error):
+                        switch error.errorCode {
+                        case .invalidInput:
+                            // Safety: This condition indicates a programming error and can only
+                            // happen if arguments to mc_tx_out_get_shared_secret are
+                            // supplied incorrectly.
+                            logger.warning("error: \(redacting: error)")
+                            return nil
+                        default:
+                            // Safety: mc_tx_out_get_shared_secret should not throw
+                            // non-documented errors.
+                            logger.warning("Unhandled LibMobileCoin error: \(redacting: error)")
+                            return nil
+                        }
+                    }
+                }
+            }
+    }
+
     static func reconstructCommitment(
         maskedValue: UInt64,
         publicKey: RistrettoPublic,
         viewPrivateKey: RistrettoPrivate
     ) -> Data32? {
-        var mcAmount = McTxOutAmount(masked_value: maskedValue)
-        return publicKey.asMcBuffer { publicKeyBufferPtr in
-            viewPrivateKey.asMcBuffer { viewPrivateKeyPtr in
-                switch Data32.make(withMcMutableBuffer: { bufferPtr, errorPtr in
-                    mc_tx_out_reconstruct_commitment(
-                        &mcAmount,
-                        publicKeyBufferPtr,
-                        viewPrivateKeyPtr,
-                        bufferPtr,
-                        &errorPtr)
-                }) {
-                case .success(let bytes):
-                    // Safety: It's safe to skip validation because
-                    // mc_tx_out_reconstruct_commitment should always return a valid
-                    // RistrettoPublic on success.
-                    return bytes as Data32
-                case .failure(let error):
-                    switch error.errorCode {
-                    case .invalidInput:
-                        // Safety: This condition indicates a programming error and can only
-                        // happen if arguments to mc_tx_out_reconstruct_commitment are
-                        // supplied incorrectly.
-                        logger.warning("error: \(redacting: error)")
-                        return nil
-                    default:
-                        // Safety: mc_tx_out_reconstruct_commitment should not throw
-                        // non-documented errors.
-                        logger.warning("Unhandled LibMobileCoin error: \(redacting: error)")
-                        return nil
+        reconstructCommitment(maskedValue: maskedValue,
+                              maskedTokenId: Data(),
```

Comment:
Does this work?  I didn't drill all the way down, but does passing an empty Data in end up defaulting to .MOB or something when attempting to pull out the tokenId from maskedTokenId?

Reply:
empty Data will be interpreted as tokenId .MOB

maybe would be better to define this as a constant somewhere to improve readability